joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/DynamicCallLinker.scala

```diff
@@ -100,6 +104,30 @@ class DynamicCallLinker(cpg: Cpg) extends CpgPass(cpg) {
     }
   }
 
+  /** Recursively returns all the super-types of the given type declaration. Does not account for circular hierarchies.
```

Comment:
If we do get a circular hierarchy here, does it just run forever? If so, I'd recommend we include a loop detector. Whenever you think something can't have loops, it ends up having loops. :D

Reply:
Will include that in the change